unstructured/ingest/connector/slack.py

```diff
@@ -0,0 +1,226 @@
+                result = self.client.conversations_history(
+                    channel=self.channel,
+                    oldest=oldest,
+                    latest=latest,
+                    cursor=result["response_metadata"]["next_cursor"],
+                )
+                messages.extend(result["messages"])
+        except SlackApiError as e:
+            logger.error(f"Error: {e}")
+
+        with open(self._tmp_download_file(), "w") as f:
+            for message in messages:
+                f.write(message["text"] + "\n")
+
+        f.close()
+
+    def write_result(self):
+        """Write the structured json result for this doc. result must be json serializable."""
+        output_filename = self._output_filename()
+        output_filename.parent.mkdir(parents=True, exist_ok=True)
+        with open(output_filename, "w") as output_f:
+            output_f.write(json.dumps(self.isd_elems_no_filename, ensure_ascii=False, indent=2))
+        logger.info(f"Wrote {output_filename}")
+
+    def convert_datetime(self, date_time):
+        date_formats = ["%Y-%m-%d", "%Y-%m-%dT%H:%M:%S", "%Y-%m-%dT%H:%M:%S%z"]
+        for format in date_formats:
+            try:
+                return datetime.strptime(date_time, format).timestamp()
+            except ValueError:
+                pass
+
+    @property
+    def filename(self):
+        """The filename of the file created from a slack channel"""
+        return self._tmp_download_file()
+
+    def cleanup_file(self):
+        """Removes the local copy the file after successful processing."""
+        if not self.config.preserve_downloads:
+            if self.config.verbose:
+                logger.info(f"cleaning up channel {self.channel}")
+            os.unlink(self._tmp_download_file())
+
+
+@requires_dependencies(dependencies=["slack_sdk"], extras="slack")
+class SlackConnector(BaseConnector):
+    """Objects of this class support fetching document(s) from"""
+
+    def __init__(self, config: SimpleSlackConfig):
+        self.config = config
+        self.cleanup_files = not config.preserve_downloads
+
+    def cleanup(self, cur_dir=None):
+        """cleanup linginering empty sub-dirs, but leave remaining files
+        (and their paths) in tact as that indicates they were not processed"""
+        if not self.cleanup_files:
+            return
+
+        if cur_dir is None:
+            cur_dir = self.config.download_dir
+        sub_dirs = os.listdir(cur_dir)
+        os.chdir(cur_dir)
+        for sub_dir in sub_dirs:
+            # don't traverse symlinks, not that there every should be any
+            if os.path.isdir(sub_dir) and not os.path.islink(sub_dir):
+                self.cleanup(sub_dir)
+        os.chdir("..")
+        if len(os.listdir(cur_dir)) == 0:
+            os.rmdir(cur_dir)
+
+    def initialize(self):
+        """Verify that can get metadata for an object, validates connections info."""
+        os.mkdir(self.config.download_dir)
```

Comment:
actually can remove this line since when writing a file on line 160, parent dirs are created if needed. (also fwiw if this dir already exists, it results an error)